spec/mongoid/persistable_spec.rb

```diff
@@ -175,24 +175,96 @@
 
         before do
           class Band
-            def my_updates
-              atomically do |d|
+            def my_updates(*args)
+              atomically(*args) do |d|
                 d.set(name: "Placebo")
                 d.unset(:origin)
               end
             end
           end
         end
 
-        let!(:update) do
-          document.atomically do |doc|
-            doc.inc(member_count: 10)
-            doc.bit(likes: { and: 13 })
-            doc.my_updates
+        context "when given join_context: false" do
+
+          let(:run_update) do
+            document.atomically do |doc|
+              doc.set origin: "Paris"
+              doc.atomically(join_context: false) do |doc2|
+                doc2.inc(member_count: 10)
+              end
+              doc.inc likes: 1
+              raise "oops"
```

Comment:
Can you please make an exception class for this like in https://github.com/mongodb/mongo-ruby-driver/blob/master/spec/integration/heartbeat_events_spec.rb#L26 and rescue this class instead of `RuntimeError` below.